### PR TITLE
Improve dead code elimination

### PR DESCRIPTION
**Pull request by zherczeg**
JIT data flow analysis has issues when the byte code contains dead code. This patch improves dead code elimination when a block has no jumps to its end.

### CODE REVIEW

File: src/parser/WASMParser.cpp [hunk 1]
```diff
@@ -1696,6 +1700,19 @@ class WASMBinaryReader : public wabt::WASMBinaryReaderDelegate {
                     m_currentFunction->m_catchInfo.push_back({ iter->m_tryStart, iter->m_tryEnd, iter->m_catchStart, stackSizeToBe, iter->m_tagIndex });
                     iter = m_catchInfo.erase(iter);
                 }
+                break;
+            }
+            case BlockInfo::Loop:
+            case BlockInfo::Block: {
+                if (blockInfo.m_byteCodeGenerationStopped && blockInfo.m_jumpToEndBrInfo.size() == 0) {
```
Comment (zherczeg) [inline src/parser/WASMParser.cpp:1707]: I assume `blockInfo.m_jumpToEndBrInfo.size()` is always zero for loops.

**Review by clover2123 — APPROVED** [review]
LGTM

**Comment by zherczeg** [comment]
Can this patch landed?

**Review by ksh8281 — APPROVED** [review]
Sorry for late!

**Pull request merged**